From e95ee4f7ad7fdb2f8c7d8abe6c73d76141f4088c Mon Sep 17 00:00:00 2001
From: Sergey Shelukhin <sershe@apache.org>
Date: Wed, 2 Dec 2015 16:16:35 -0800
Subject: [PATCH 648/723] CDH-43369: HIVE-12556 : Ctrl-C in beeline doesn't
 kill Tez query on HS2 (Sergey Shelukhin, reviewed
 by Gunther Hagleitner)

Only backport relate to HIVE-14596 and exclude tez only changes.
Conflicts:
	ql/src/java/org/apache/hadoop/hive/ql/exec/tez/TezJobMonitor.java
	ql/src/java/org/apache/hadoop/hive/ql/exec/tez/TezTask.java

Change-Id: I9dc9391c730c2cdb76069bbaf59547ca59c7e4ed
(cherry picked from commit 3859d5bcd505cf304cfbedb188afb3141effe769)
---
 ql/src/java/org/apache/hadoop/hive/ql/Driver.java |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/Driver.java b/ql/src/java/org/apache/hadoop/hive/ql/Driver.java
index fcbba5e..36f6ebb 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/Driver.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/Driver.java
@@ -1869,6 +1869,11 @@ public void setTryCount(int tryCount) {
 
   public int close() {
     try {
+      try {
+        releaseResources();
+      } catch (Exception e) {
+        LOG.info("Exception while releasing resources", e);
+      }
       if (fetchTask != null) {
         try {
           fetchTask.clearFetch();
-- 
1.7.9.5

